pkg/streaming/util/message/adaptor/message.go

```diff
@@ -0,0 +1,173 @@
+package adaptor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/commonpb"
+	"github.com/milvus-io/milvus-proto/go-api/v2/msgpb"
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/pkg/mq/msgstream"
+	"github.com/milvus-io/milvus/pkg/streaming/util/message"
+	"github.com/milvus-io/milvus/pkg/util/commonpbutil"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+)
+
+var unmashalerDispatcher = (&msgstream.ProtoUDFactory{}).NewUnmarshalDispatcher()
+
+// FromMessageToMsgPack converts message to msgpack.
+// Same TimeTick must be sent with same msgpack.
+// !!! Msgs must be keep same time tick.
+// TODO: remove this function after remove the msgstream implementation.
+func NewMsgPackFromMessage(msgs ...message.ImmutableMessage) (*msgstream.MsgPack, error) {
+	if len(msgs) == 0 {
+		return nil, nil
+	}
+	allTsMsgs := make([]msgstream.TsMsg, 0, len(msgs))
+
+	var finalErr error
+	for _, msg := range msgs {
+		var tsMsgs []msgstream.TsMsg
+		var err error
+		switch msg.Version() {
+		case message.VersionOld:
+			tsMsgs, err = fromMessageToTsMsgVOld(msg)
+		case message.VersionV1:
+			tsMsgs, err = fromMessageToTsMsgV1(msg)
+		default:
+			panic("unsupported message version")
+		}
+		if err != nil {
+			finalErr = errors.CombineErrors(finalErr, errors.Wrapf(err, "Failed to convert message to msgpack, %v", msg.MessageID()))
```

Comment:
Return immediately if the error is not nil; checking len(allTsMsgs) == 0 may skip some illegal messages at line 48.

Reply:
no, we only filter out illegal message, but keep the remaining legal message to consume.

Reply:
No illegal messages are expected. Log or return an error when filtering out illegal messages; only return an error if all messages are illegal. It is confusing that you shouldn't return an error on line 43 when all messages are illegal if the aim is to skip those messages?